src/gtc/common.py

```diff
@@ -336,13 +336,21 @@ def zero(cls) -> "CartesianOffset":
     def to_dict(self) -> Dict[str, int]:
         return {"i": self.i, "j": self.j, "k": self.k}
 
+    def to_tuple(self) -> Tuple[int, int, int]:
+        return self.i, self.j, self.k
+
 
 class VariableKOffset(GenericNode, Generic[ExprT]):
+    i: int = 0
+    j: int = 0
```

Comment:
I don't like this change. `VariableKOffset` cannot have `i` or `j` offsets so it is wrong to add them here. If it's a matter of convenience then we should solve the root of the problem.

Reply:
I changed it to be a property rather than a class attribute. In this way, we can continue to just assume that any offset, cartesian or variable, will have attributes `i`, `j` and `k`.